2020-03-25 11:54:22  [ hubg-00074267-9a84-4bb5-b771-b95b66b62746-StreamThread-1:837 ] - [ ERROR ]  stream-thread [main] Unexpected error during topic creation for hubg-counts-store-repartition.
Error message was: org.apache.kafka.common.errors.UnsupportedVersionException: The broker does not support CREATE_TOPICS
2020-03-25 11:54:22  [ hubg-00074267-9a84-4bb5-b771-b95b66b62746-StreamThread-1:840 ] - [ ERROR ]  stream-thread [hubg-00074267-9a84-4bb5-b771-b95b66b62746-StreamThread-1] Encountered the following unexpected Kafka exception during processing, this usually indicate Streams internal errors:
org.apache.kafka.streams.errors.StreamsException: Could not create topic hubg-counts-store-repartition.
	at org.apache.kafka.streams.processor.internals.InternalTopicManager.makeReady(InternalTopicManager.java:148)
	at org.apache.kafka.streams.processor.internals.StreamsPartitionAssignor.prepareTopic(StreamsPartitionAssignor.java:971)
	at org.apache.kafka.streams.processor.internals.StreamsPartitionAssignor.assign(StreamsPartitionAssignor.java:528)
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.performAssignment(ConsumerCoordinator.java:424)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.onJoinLeader(AbstractCoordinator.java:622)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.access$1100(AbstractCoordinator.java:107)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$JoinGroupResponseHandler.handle(AbstractCoordinator.java:544)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$JoinGroupResponseHandler.handle(AbstractCoordinator.java:527)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:978)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:958)
	at org.apache.kafka.clients.consumer.internals.RequestFuture$1.onSuccess(RequestFuture.java:204)
	at org.apache.kafka.clients.consumer.internals.RequestFuture.fireSuccess(RequestFuture.java:167)
	at org.apache.kafka.clients.consumer.internals.RequestFuture.complete(RequestFuture.java:127)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient$RequestFutureCompletionHandler.fireCompletion(ConsumerNetworkClient.java:578)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.firePendingCompletedRequests(ConsumerNetworkClient.java:388)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:294)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:233)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:212)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:415)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:358)
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:353)
	at org.apache.kafka.clients.consumer.KafkaConsumer.updateAssignmentMetadataIfNeeded(KafkaConsumer.java:1251)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1216)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1201)
	at org.apache.kafka.streams.processor.internals.StreamThread.pollRequests(StreamThread.java:941)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:846)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
Caused by: org.apache.kafka.common.errors.UnsupportedVersionException: The broker does not support CREATE_TOPICS
2020-03-25 11:54:22  [ hubg-00074267-9a84-4bb5-b771-b95b66b62746-StreamThread-1:852 ] - [ ERROR ]  stream-client [hubg-00074267-9a84-4bb5-b771-b95b66b62746] All stream threads have died. The instance will be in error state and should be closed.
2020-03-25 11:55:16  [ hubg-e3584dce-3938-4d6a-8cde-fc7d49a932a8-StreamThread-1:921 ] - [ ERROR ]  stream-thread [main] Unexpected error during topic creation for hubg-counts-store-changelog.
Error message was: org.apache.kafka.common.errors.UnsupportedVersionException: The broker does not support CREATE_TOPICS
2020-03-25 11:55:16  [ hubg-e3584dce-3938-4d6a-8cde-fc7d49a932a8-StreamThread-1:922 ] - [ ERROR ]  stream-thread [hubg-e3584dce-3938-4d6a-8cde-fc7d49a932a8-StreamThread-1] Encountered the following unexpected Kafka exception during processing, this usually indicate Streams internal errors:
org.apache.kafka.streams.errors.StreamsException: Could not create topic hubg-counts-store-changelog.
	at org.apache.kafka.streams.processor.internals.InternalTopicManager.makeReady(InternalTopicManager.java:148)
	at org.apache.kafka.streams.processor.internals.StreamsPartitionAssignor.prepareTopic(StreamsPartitionAssignor.java:971)
	at org.apache.kafka.streams.processor.internals.StreamsPartitionAssignor.assign(StreamsPartitionAssignor.java:618)
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.performAssignment(ConsumerCoordinator.java:424)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.onJoinLeader(AbstractCoordinator.java:622)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.access$1100(AbstractCoordinator.java:107)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$JoinGroupResponseHandler.handle(AbstractCoordinator.java:544)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$JoinGroupResponseHandler.handle(AbstractCoordinator.java:527)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:978)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:958)
	at org.apache.kafka.clients.consumer.internals.RequestFuture$1.onSuccess(RequestFuture.java:204)
	at org.apache.kafka.clients.consumer.internals.RequestFuture.fireSuccess(RequestFuture.java:167)
	at org.apache.kafka.clients.consumer.internals.RequestFuture.complete(RequestFuture.java:127)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient$RequestFutureCompletionHandler.fireCompletion(ConsumerNetworkClient.java:578)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.firePendingCompletedRequests(ConsumerNetworkClient.java:388)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:294)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:233)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:212)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:415)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:358)
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:353)
	at org.apache.kafka.clients.consumer.KafkaConsumer.updateAssignmentMetadataIfNeeded(KafkaConsumer.java:1251)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1216)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1201)
	at org.apache.kafka.streams.processor.internals.StreamThread.pollRequests(StreamThread.java:941)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:850)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
Caused by: org.apache.kafka.common.errors.UnsupportedVersionException: The broker does not support CREATE_TOPICS
2020-03-25 11:55:16  [ hubg-e3584dce-3938-4d6a-8cde-fc7d49a932a8-StreamThread-1:932 ] - [ ERROR ]  stream-client [hubg-e3584dce-3938-4d6a-8cde-fc7d49a932a8] All stream threads have died. The instance will be in error state and should be closed.
2020-03-25 12:25:45  [ hubg-bcbad491-b9a3-451d-a36d-75ea1d08a087-CleanupThread:1800727 ] - [ ERROR ]  stream-thread [hubg-bcbad491-b9a3-451d-a36d-75ea1d08a087-CleanupThread] Failed to delete the state directory.
java.nio.file.DirectoryNotEmptyException: \tmp\kafka-streams\hubg\0_0
	at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:266)
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
	at java.nio.file.Files.delete(Files.java:1126)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:769)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:752)
	at java.nio.file.Files.walkFileTree(Files.java:2688)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at org.apache.kafka.common.utils.Utils.delete(Utils.java:752)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:301)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:264)
	at org.apache.kafka.streams.KafkaStreams.lambda$start$1(KafkaStreams.java:802)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
2020-03-25 12:55:45  [ hubg-bcbad491-b9a3-451d-a36d-75ea1d08a087-CleanupThread:3600726 ] - [ ERROR ]  stream-thread [hubg-bcbad491-b9a3-451d-a36d-75ea1d08a087-CleanupThread] Failed to delete the state directory.
java.nio.file.DirectoryNotEmptyException: \tmp\kafka-streams\hubg\0_0
	at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:266)
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
	at java.nio.file.Files.delete(Files.java:1126)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:769)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:752)
	at java.nio.file.Files.walkFileTree(Files.java:2688)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at org.apache.kafka.common.utils.Utils.delete(Utils.java:752)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:301)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:264)
	at org.apache.kafka.streams.KafkaStreams.lambda$start$1(KafkaStreams.java:802)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
2020-03-25 13:15:45  [ hubg-bcbad491-b9a3-451d-a36d-75ea1d08a087-CleanupThread:4800728 ] - [ ERROR ]  stream-thread [hubg-bcbad491-b9a3-451d-a36d-75ea1d08a087-CleanupThread] Failed to delete the state directory.
java.nio.file.DirectoryNotEmptyException: \tmp\kafka-streams\hubg\0_0
	at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:266)
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
	at java.nio.file.Files.delete(Files.java:1126)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:769)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:752)
	at java.nio.file.Files.walkFileTree(Files.java:2688)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at org.apache.kafka.common.utils.Utils.delete(Utils.java:752)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:301)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:264)
	at org.apache.kafka.streams.KafkaStreams.lambda$start$1(KafkaStreams.java:802)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
2020-03-25 13:45:45  [ hubg-bcbad491-b9a3-451d-a36d-75ea1d08a087-CleanupThread:6600728 ] - [ ERROR ]  stream-thread [hubg-bcbad491-b9a3-451d-a36d-75ea1d08a087-CleanupThread] Failed to delete the state directory.
java.nio.file.DirectoryNotEmptyException: \tmp\kafka-streams\hubg\0_0
	at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:266)
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
	at java.nio.file.Files.delete(Files.java:1126)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:769)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:752)
	at java.nio.file.Files.walkFileTree(Files.java:2688)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at org.apache.kafka.common.utils.Utils.delete(Utils.java:752)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:301)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:264)
	at org.apache.kafka.streams.KafkaStreams.lambda$start$1(KafkaStreams.java:802)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
2020-03-25 14:15:45  [ hubg-bcbad491-b9a3-451d-a36d-75ea1d08a087-CleanupThread:8400729 ] - [ ERROR ]  stream-thread [hubg-bcbad491-b9a3-451d-a36d-75ea1d08a087-CleanupThread] Failed to delete the state directory.
java.nio.file.DirectoryNotEmptyException: \tmp\kafka-streams\hubg\0_0
	at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:266)
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
	at java.nio.file.Files.delete(Files.java:1126)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:769)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:752)
	at java.nio.file.Files.walkFileTree(Files.java:2688)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at org.apache.kafka.common.utils.Utils.delete(Utils.java:752)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:301)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:264)
	at org.apache.kafka.streams.KafkaStreams.lambda$start$1(KafkaStreams.java:802)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
2020-03-25 14:45:45  [ hubg-bcbad491-b9a3-451d-a36d-75ea1d08a087-CleanupThread:10200726 ] - [ ERROR ]  stream-thread [hubg-bcbad491-b9a3-451d-a36d-75ea1d08a087-CleanupThread] Failed to delete the state directory.
java.nio.file.DirectoryNotEmptyException: \tmp\kafka-streams\hubg\0_0
	at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:266)
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
	at java.nio.file.Files.delete(Files.java:1126)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:769)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:752)
	at java.nio.file.Files.walkFileTree(Files.java:2688)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at org.apache.kafka.common.utils.Utils.delete(Utils.java:752)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:301)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:264)
	at org.apache.kafka.streams.KafkaStreams.lambda$start$1(KafkaStreams.java:802)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
2020-03-25 16:24:58  [ streams-temperature-e990879c-c886-455e-a932-356296a19435-StreamThread-1:2324 ] - [ ERROR ]  stream-thread [main] Unexpected error during topic creation for streams-temperature-KSTREAM-REDUCE-STATE-STORE-0000000002-repartition.
Error message was: org.apache.kafka.common.errors.UnsupportedVersionException: The broker does not support CREATE_TOPICS
2020-03-25 16:24:58  [ streams-temperature-e990879c-c886-455e-a932-356296a19435-StreamThread-1:2326 ] - [ ERROR ]  stream-thread [streams-temperature-e990879c-c886-455e-a932-356296a19435-StreamThread-1] Encountered the following unexpected Kafka exception during processing, this usually indicate Streams internal errors:
org.apache.kafka.streams.errors.StreamsException: Could not create topic streams-temperature-KSTREAM-REDUCE-STATE-STORE-0000000002-repartition.
	at org.apache.kafka.streams.processor.internals.InternalTopicManager.makeReady(InternalTopicManager.java:148)
	at org.apache.kafka.streams.processor.internals.StreamsPartitionAssignor.prepareTopic(StreamsPartitionAssignor.java:971)
	at org.apache.kafka.streams.processor.internals.StreamsPartitionAssignor.assign(StreamsPartitionAssignor.java:528)
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.performAssignment(ConsumerCoordinator.java:424)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.onJoinLeader(AbstractCoordinator.java:622)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.access$1100(AbstractCoordinator.java:107)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$JoinGroupResponseHandler.handle(AbstractCoordinator.java:544)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$JoinGroupResponseHandler.handle(AbstractCoordinator.java:527)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:978)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:958)
	at org.apache.kafka.clients.consumer.internals.RequestFuture$1.onSuccess(RequestFuture.java:204)
	at org.apache.kafka.clients.consumer.internals.RequestFuture.fireSuccess(RequestFuture.java:167)
	at org.apache.kafka.clients.consumer.internals.RequestFuture.complete(RequestFuture.java:127)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient$RequestFutureCompletionHandler.fireCompletion(ConsumerNetworkClient.java:578)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.firePendingCompletedRequests(ConsumerNetworkClient.java:388)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:294)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:233)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:212)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:415)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:358)
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:353)
	at org.apache.kafka.clients.consumer.KafkaConsumer.updateAssignmentMetadataIfNeeded(KafkaConsumer.java:1251)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1216)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1201)
	at org.apache.kafka.streams.processor.internals.StreamThread.pollRequests(StreamThread.java:941)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:846)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
Caused by: org.apache.kafka.common.errors.UnsupportedVersionException: The broker does not support CREATE_TOPICS
2020-03-25 16:24:58  [ streams-temperature-e990879c-c886-455e-a932-356296a19435-StreamThread-1:2347 ] - [ ERROR ]  stream-client [streams-temperature-e990879c-c886-455e-a932-356296a19435] All stream threads have died. The instance will be in error state and should be closed.
2020-03-25 16:25:44  [ streams-temperature-d1f76380-5681-43dc-9d68-22e34482d849-StreamThread-1:1120 ] - [ ERROR ]  stream-thread [main] Unexpected error during topic creation for streams-temperature-KSTREAM-REDUCE-STATE-STORE-0000000002-changelog.
Error message was: org.apache.kafka.common.errors.UnsupportedVersionException: The broker does not support CREATE_TOPICS
2020-03-25 16:25:44  [ streams-temperature-d1f76380-5681-43dc-9d68-22e34482d849-StreamThread-1:1121 ] - [ ERROR ]  stream-thread [streams-temperature-d1f76380-5681-43dc-9d68-22e34482d849-StreamThread-1] Encountered the following unexpected Kafka exception during processing, this usually indicate Streams internal errors:
org.apache.kafka.streams.errors.StreamsException: Could not create topic streams-temperature-KSTREAM-REDUCE-STATE-STORE-0000000002-changelog.
	at org.apache.kafka.streams.processor.internals.InternalTopicManager.makeReady(InternalTopicManager.java:148)
	at org.apache.kafka.streams.processor.internals.StreamsPartitionAssignor.prepareTopic(StreamsPartitionAssignor.java:971)
	at org.apache.kafka.streams.processor.internals.StreamsPartitionAssignor.assign(StreamsPartitionAssignor.java:618)
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.performAssignment(ConsumerCoordinator.java:424)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.onJoinLeader(AbstractCoordinator.java:622)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.access$1100(AbstractCoordinator.java:107)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$JoinGroupResponseHandler.handle(AbstractCoordinator.java:544)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$JoinGroupResponseHandler.handle(AbstractCoordinator.java:527)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:978)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator$CoordinatorResponseHandler.onSuccess(AbstractCoordinator.java:958)
	at org.apache.kafka.clients.consumer.internals.RequestFuture$1.onSuccess(RequestFuture.java:204)
	at org.apache.kafka.clients.consumer.internals.RequestFuture.fireSuccess(RequestFuture.java:167)
	at org.apache.kafka.clients.consumer.internals.RequestFuture.complete(RequestFuture.java:127)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient$RequestFutureCompletionHandler.fireCompletion(ConsumerNetworkClient.java:578)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.firePendingCompletedRequests(ConsumerNetworkClient.java:388)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:294)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:233)
	at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:212)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.joinGroupIfNeeded(AbstractCoordinator.java:415)
	at org.apache.kafka.clients.consumer.internals.AbstractCoordinator.ensureActiveGroup(AbstractCoordinator.java:358)
	at org.apache.kafka.clients.consumer.internals.ConsumerCoordinator.poll(ConsumerCoordinator.java:353)
	at org.apache.kafka.clients.consumer.KafkaConsumer.updateAssignmentMetadataIfNeeded(KafkaConsumer.java:1251)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1216)
	at org.apache.kafka.clients.consumer.KafkaConsumer.poll(KafkaConsumer.java:1201)
	at org.apache.kafka.streams.processor.internals.StreamThread.pollRequests(StreamThread.java:941)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:850)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
Caused by: org.apache.kafka.common.errors.UnsupportedVersionException: The broker does not support CREATE_TOPICS
2020-03-25 16:25:44  [ streams-temperature-d1f76380-5681-43dc-9d68-22e34482d849-StreamThread-1:1132 ] - [ ERROR ]  stream-client [streams-temperature-d1f76380-5681-43dc-9d68-22e34482d849] All stream threads have died. The instance will be in error state and should be closed.
2020-03-25 16:29:05  [ streams-temperature-6bd57e6b-ef43-4c5c-857b-8a84a54781f7-StreamThread-1:72421 ] - [ ERROR ]  stream-thread [streams-temperature-6bd57e6b-ef43-4c5c-857b-8a84a54781f7-StreamThread-1] Failed to process stream task 1_0 due to the following error:
java.lang.NullPointerException
	at com.yixin.hubg.TemperatureDemo$1.test(TemperatureDemo.java:73)
	at com.yixin.hubg.TemperatureDemo$1.test(TemperatureDemo.java:67)
	at org.apache.kafka.streams.kstream.internals.KStreamFilter$KStreamFilterProcessor.process(KStreamFilter.java:42)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.kstream.internals.KStreamMapValues$KStreamMapProcessor.process(KStreamMapValues.java:41)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.kstream.internals.TimestampedTupleForwarder.maybeForward(TimestampedTupleForwarder.java:60)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:139)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 16:29:05  [ streams-temperature-6bd57e6b-ef43-4c5c-857b-8a84a54781f7-StreamThread-1:72423 ] - [ ERROR ]  stream-thread [streams-temperature-6bd57e6b-ef43-4c5c-857b-8a84a54781f7-StreamThread-1] Encountered the following error during processing:
java.lang.NullPointerException
	at com.yixin.hubg.TemperatureDemo$1.test(TemperatureDemo.java:73)
	at com.yixin.hubg.TemperatureDemo$1.test(TemperatureDemo.java:67)
	at org.apache.kafka.streams.kstream.internals.KStreamFilter$KStreamFilterProcessor.process(KStreamFilter.java:42)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.kstream.internals.KStreamMapValues$KStreamMapProcessor.process(KStreamMapValues.java:41)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.kstream.internals.TimestampedTupleForwarder.maybeForward(TimestampedTupleForwarder.java:60)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:139)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 16:29:05  [ streams-temperature-6bd57e6b-ef43-4c5c-857b-8a84a54781f7-StreamThread-1:72462 ] - [ ERROR ]  stream-client [streams-temperature-6bd57e6b-ef43-4c5c-857b-8a84a54781f7] All stream threads have died. The instance will be in error state and should be closed.
2020-03-25 17:19:34  [ streams-temperature-39fbcf84-5885-4683-a343-5205a6bcd471-StreamThread-1:1802 ] - [ ERROR ]  stream-thread [streams-temperature-39fbcf84-5885-4683-a343-5205a6bcd471-StreamThread-1] Failed to process stream task 1_0 due to the following error:
java.lang.NullPointerException
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:57)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:51)
	at org.apache.kafka.streams.kstream.internals.TimeWindowedKStreamImpl.lambda$aggregatorForReducer$0(TimeWindowedKStreamImpl.java:223)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:135)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 17:19:34  [ streams-temperature-39fbcf84-5885-4683-a343-5205a6bcd471-StreamThread-1:1804 ] - [ ERROR ]  stream-thread [streams-temperature-39fbcf84-5885-4683-a343-5205a6bcd471-StreamThread-1] Encountered the following error during processing:
java.lang.NullPointerException
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:57)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:51)
	at org.apache.kafka.streams.kstream.internals.TimeWindowedKStreamImpl.lambda$aggregatorForReducer$0(TimeWindowedKStreamImpl.java:223)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:135)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 17:19:34  [ streams-temperature-39fbcf84-5885-4683-a343-5205a6bcd471-StreamThread-1:1840 ] - [ ERROR ]  stream-client [streams-temperature-39fbcf84-5885-4683-a343-5205a6bcd471] All stream threads have died. The instance will be in error state and should be closed.
2020-03-25 17:25:46  [ streams-temperature-9309bfa7-71d5-464d-9148-1ad6c7b9bf9f-StreamThread-1:1599 ] - [ ERROR ]  stream-thread [streams-temperature-9309bfa7-71d5-464d-9148-1ad6c7b9bf9f-StreamThread-1] Failed to process stream task 1_0 due to the following error:
java.lang.NullPointerException
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:57)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:51)
	at org.apache.kafka.streams.kstream.internals.TimeWindowedKStreamImpl.lambda$aggregatorForReducer$0(TimeWindowedKStreamImpl.java:223)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:135)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 17:25:46  [ streams-temperature-9309bfa7-71d5-464d-9148-1ad6c7b9bf9f-StreamThread-1:1601 ] - [ ERROR ]  stream-thread [streams-temperature-9309bfa7-71d5-464d-9148-1ad6c7b9bf9f-StreamThread-1] Encountered the following error during processing:
java.lang.NullPointerException
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:57)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:51)
	at org.apache.kafka.streams.kstream.internals.TimeWindowedKStreamImpl.lambda$aggregatorForReducer$0(TimeWindowedKStreamImpl.java:223)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:135)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 17:25:46  [ streams-temperature-9309bfa7-71d5-464d-9148-1ad6c7b9bf9f-StreamThread-1:1633 ] - [ ERROR ]  stream-client [streams-temperature-9309bfa7-71d5-464d-9148-1ad6c7b9bf9f] All stream threads have died. The instance will be in error state and should be closed.
2020-03-25 17:45:15  [ streams-temperature-75ad90bf-37b7-46b9-bc2e-ed8ffd02e895-StreamThread-1:1690 ] - [ ERROR ]  stream-thread [streams-temperature-75ad90bf-37b7-46b9-bc2e-ed8ffd02e895-StreamThread-1] Failed to process stream task 1_0 due to the following error:
java.lang.NumberFormatException: For input string: "{“temp”:19, “humidity”: 25}"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:58)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:51)
	at org.apache.kafka.streams.kstream.internals.TimeWindowedKStreamImpl.lambda$aggregatorForReducer$0(TimeWindowedKStreamImpl.java:223)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:135)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 17:45:15  [ streams-temperature-75ad90bf-37b7-46b9-bc2e-ed8ffd02e895-StreamThread-1:1693 ] - [ ERROR ]  stream-thread [streams-temperature-75ad90bf-37b7-46b9-bc2e-ed8ffd02e895-StreamThread-1] Encountered the following error during processing:
java.lang.NumberFormatException: For input string: "{“temp”:19, “humidity”: 25}"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:58)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:51)
	at org.apache.kafka.streams.kstream.internals.TimeWindowedKStreamImpl.lambda$aggregatorForReducer$0(TimeWindowedKStreamImpl.java:223)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:135)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 17:45:15  [ streams-temperature-75ad90bf-37b7-46b9-bc2e-ed8ffd02e895-StreamThread-1:1725 ] - [ ERROR ]  stream-client [streams-temperature-75ad90bf-37b7-46b9-bc2e-ed8ffd02e895] All stream threads have died. The instance will be in error state and should be closed.
2020-03-25 17:46:39  [ streams-temperature-a2c2ee7c-6762-43a8-97a6-c10a8a53bd84-StreamThread-1:1627 ] - [ ERROR ]  stream-thread [streams-temperature-a2c2ee7c-6762-43a8-97a6-c10a8a53bd84-StreamThread-1] Failed to process stream task 1_0 due to the following error:
java.lang.NumberFormatException: For input string: "{“temp”:19, “humidity”: 25}"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:58)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:51)
	at org.apache.kafka.streams.kstream.internals.TimeWindowedKStreamImpl.lambda$aggregatorForReducer$0(TimeWindowedKStreamImpl.java:223)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:135)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 17:46:39  [ streams-temperature-a2c2ee7c-6762-43a8-97a6-c10a8a53bd84-StreamThread-1:1629 ] - [ ERROR ]  stream-thread [streams-temperature-a2c2ee7c-6762-43a8-97a6-c10a8a53bd84-StreamThread-1] Encountered the following error during processing:
java.lang.NumberFormatException: For input string: "{“temp”:19, “humidity”: 25}"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:58)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:51)
	at org.apache.kafka.streams.kstream.internals.TimeWindowedKStreamImpl.lambda$aggregatorForReducer$0(TimeWindowedKStreamImpl.java:223)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:135)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 17:46:39  [ streams-temperature-a2c2ee7c-6762-43a8-97a6-c10a8a53bd84-StreamThread-1:1657 ] - [ ERROR ]  stream-client [streams-temperature-a2c2ee7c-6762-43a8-97a6-c10a8a53bd84] All stream threads have died. The instance will be in error state and should be closed.
2020-03-25 17:49:01  [ streams-temperature-391f9465-7eb9-40c4-a565-6e0dec0a5051-StreamThread-1:1601 ] - [ ERROR ]  stream-thread [streams-temperature-391f9465-7eb9-40c4-a565-6e0dec0a5051-StreamThread-1] Failed to process stream task 1_0 due to the following error:
java.lang.NumberFormatException: For input string: "{“temp”:19, “humidity”: 25}"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:59)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:51)
	at org.apache.kafka.streams.kstream.internals.TimeWindowedKStreamImpl.lambda$aggregatorForReducer$0(TimeWindowedKStreamImpl.java:223)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:135)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 17:49:01  [ streams-temperature-391f9465-7eb9-40c4-a565-6e0dec0a5051-StreamThread-1:1604 ] - [ ERROR ]  stream-thread [streams-temperature-391f9465-7eb9-40c4-a565-6e0dec0a5051-StreamThread-1] Encountered the following error during processing:
java.lang.NumberFormatException: For input string: "{“temp”:19, “humidity”: 25}"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:59)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:51)
	at org.apache.kafka.streams.kstream.internals.TimeWindowedKStreamImpl.lambda$aggregatorForReducer$0(TimeWindowedKStreamImpl.java:223)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:135)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 17:49:01  [ streams-temperature-391f9465-7eb9-40c4-a565-6e0dec0a5051-StreamThread-1:1635 ] - [ ERROR ]  stream-client [streams-temperature-391f9465-7eb9-40c4-a565-6e0dec0a5051] All stream threads have died. The instance will be in error state and should be closed.
2020-03-25 17:50:27  [ streams-temperature-ed26935e-5d70-4991-ba87-161f3d1660fc-StreamThread-1:1646 ] - [ ERROR ]  stream-thread [streams-temperature-ed26935e-5d70-4991-ba87-161f3d1660fc-StreamThread-1] Failed to process stream task 1_0 due to the following error:
java.lang.NullPointerException
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:58)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:51)
	at org.apache.kafka.streams.kstream.internals.TimeWindowedKStreamImpl.lambda$aggregatorForReducer$0(TimeWindowedKStreamImpl.java:223)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:135)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 17:50:27  [ streams-temperature-ed26935e-5d70-4991-ba87-161f3d1660fc-StreamThread-1:1648 ] - [ ERROR ]  stream-thread [streams-temperature-ed26935e-5d70-4991-ba87-161f3d1660fc-StreamThread-1] Encountered the following error during processing:
java.lang.NullPointerException
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:58)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:51)
	at org.apache.kafka.streams.kstream.internals.TimeWindowedKStreamImpl.lambda$aggregatorForReducer$0(TimeWindowedKStreamImpl.java:223)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:135)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 17:50:27  [ streams-temperature-ed26935e-5d70-4991-ba87-161f3d1660fc-StreamThread-1:1679 ] - [ ERROR ]  stream-client [streams-temperature-ed26935e-5d70-4991-ba87-161f3d1660fc] All stream threads have died. The instance will be in error state and should be closed.
2020-03-25 17:53:19  [ streams-temperature-fb49c35e-077d-4a28-9dd2-e2ba27dc1319-StreamThread-1:1622 ] - [ ERROR ]  stream-thread [streams-temperature-fb49c35e-077d-4a28-9dd2-e2ba27dc1319-StreamThread-1] Failed to process stream task 1_0 due to the following error:
java.lang.NumberFormatException: For input string: "{"temp":19, "humidity": 25}"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:59)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:51)
	at org.apache.kafka.streams.kstream.internals.TimeWindowedKStreamImpl.lambda$aggregatorForReducer$0(TimeWindowedKStreamImpl.java:223)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:135)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 17:53:19  [ streams-temperature-fb49c35e-077d-4a28-9dd2-e2ba27dc1319-StreamThread-1:1625 ] - [ ERROR ]  stream-thread [streams-temperature-fb49c35e-077d-4a28-9dd2-e2ba27dc1319-StreamThread-1] Encountered the following error during processing:
java.lang.NumberFormatException: For input string: "{"temp":19, "humidity": 25}"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:580)
	at java.lang.Integer.parseInt(Integer.java:615)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:59)
	at com.yixin.hubg.TemperatureDemo$2.apply(TemperatureDemo.java:51)
	at org.apache.kafka.streams.kstream.internals.TimeWindowedKStreamImpl.lambda$aggregatorForReducer$0(TimeWindowedKStreamImpl.java:223)
	at org.apache.kafka.streams.kstream.internals.KStreamWindowAggregate$KStreamWindowAggregateProcessor.process(KStreamWindowAggregate.java:135)
	at org.apache.kafka.streams.processor.internals.ProcessorNode.process(ProcessorNode.java:117)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:201)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:180)
	at org.apache.kafka.streams.processor.internals.ProcessorContextImpl.forward(ProcessorContextImpl.java:133)
	at org.apache.kafka.streams.processor.internals.SourceNode.process(SourceNode.java:87)
	at org.apache.kafka.streams.processor.internals.StreamTask.process(StreamTask.java:366)
	at org.apache.kafka.streams.processor.internals.AssignedStreamsTasks.process(AssignedStreamsTasks.java:199)
	at org.apache.kafka.streams.processor.internals.TaskManager.process(TaskManager.java:420)
	at org.apache.kafka.streams.processor.internals.StreamThread.runOnce(StreamThread.java:890)
	at org.apache.kafka.streams.processor.internals.StreamThread.runLoop(StreamThread.java:805)
	at org.apache.kafka.streams.processor.internals.StreamThread.run(StreamThread.java:774)
2020-03-25 17:53:19  [ streams-temperature-fb49c35e-077d-4a28-9dd2-e2ba27dc1319-StreamThread-1:1662 ] - [ ERROR ]  stream-client [streams-temperature-fb49c35e-077d-4a28-9dd2-e2ba27dc1319] All stream threads have died. The instance will be in error state and should be closed.
2020-03-25 18:31:55  [ streams-temperature-3972cfa5-e303-422d-bafb-cc20e7ab1f9c-CleanupThread:1801173 ] - [ ERROR ]  stream-thread [streams-temperature-3972cfa5-e303-422d-bafb-cc20e7ab1f9c-CleanupThread] Failed to delete the state directory.
java.nio.file.DirectoryNotEmptyException: \tmp\kafka-streams\streams-temperature\0_0
	at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:266)
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
	at java.nio.file.Files.delete(Files.java:1126)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:769)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:752)
	at java.nio.file.Files.walkFileTree(Files.java:2688)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at org.apache.kafka.common.utils.Utils.delete(Utils.java:752)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:301)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:264)
	at org.apache.kafka.streams.KafkaStreams.lambda$start$1(KafkaStreams.java:802)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
2020-03-25 19:05:47  [ streams-temperature-4ce58d60-87df-4e9a-a23c-756e17616a56-CleanupThread:1200777 ] - [ ERROR ]  stream-thread [streams-temperature-4ce58d60-87df-4e9a-a23c-756e17616a56-CleanupThread] Failed to delete the state directory.
java.nio.file.DirectoryNotEmptyException: \tmp\kafka-streams\streams-temperature\0_0
	at sun.nio.fs.WindowsFileSystemProvider.implDelete(WindowsFileSystemProvider.java:266)
	at sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:103)
	at java.nio.file.Files.delete(Files.java:1126)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:769)
	at org.apache.kafka.common.utils.Utils$2.postVisitDirectory(Utils.java:752)
	at java.nio.file.Files.walkFileTree(Files.java:2688)
	at java.nio.file.Files.walkFileTree(Files.java:2742)
	at org.apache.kafka.common.utils.Utils.delete(Utils.java:752)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:301)
	at org.apache.kafka.streams.processor.internals.StateDirectory.cleanRemovedTasks(StateDirectory.java:264)
	at org.apache.kafka.streams.KafkaStreams.lambda$start$1(KafkaStreams.java:802)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
